runtime/src/main/java/org/corfudb/runtime/exceptions/UnrecoverableStreamingException.java

```diff
@@ -0,0 +1,15 @@
+package org.corfudb.runtime.exceptions;
+
+/**
+ * CorfuStore Streaming Exceptions are wrapped in this Exception type, so clients
+ * can distinguish from streaming exceptions caused by their own processing logic
+ * and those internal to Corfu, which will require a snapshot in order to recover.
+ *
+ * @author annym
+ */
+public class UnrecoverableStreamingException extends RuntimeException {
```

Comment:
Can you just use `StreamSubscriptionException`, just to keep the API's interface minimal.

Reply:
I used that one in my first version, but then it was sort of confusing because this has nothing to do with an exception upon subscriptio. Probably we can rename that one to UnrecoverableStreamingException? so its more generic..

Reply:
`Unrecoverable` convey that there was a hard fault that cannot be recovered, which is not true. Perhaps, `StreamingException`.